src/TransitionSystems/composition.rs

```diff
@@ -22,6 +22,15 @@ pub struct Composition {
     dim: ClockIndex,
 }
 
+pub enum CompositionResult{
+    Success,
+    Failure(CompositionFailure)
+}
+
+pub enum CompositionFailure {
+    OutputsNotDisjoint(Composition)
```

Comment:
Tænker ikke at hele composition er nødvendig - det burde være nok med navn på de to components, der indgår , samt de input/output der ikke er disjoint